java/server/src/com/appdynamics/wpt/WptHookAwareHandler.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) AppDynamics Inc
+ * All rights reserved
+ */
+package com.appdynamics.wpt;
+
+import org.openqa.selenium.remote.DriverCommand;
+import org.openqa.selenium.remote.Command;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+import java.net.MalformedURLException;
+
+/**
+ * @author <a mailto="renault.lecoultre@appdynamics.com">Renault John Lecoultre</a>
+ * @since 6/26/2015
+ *
+ * @author <a mailto="karthik3186@gmail.com">Karthik Krishnamurthy</a>
+ * @since 2/14/2015
+ */
+public class WptHookAwareHandler {
+  private WptHookClient wptHookClient;
+  private final Logger log;
+
+  public WptHookAwareHandler(Logger log, String url) {
+    this.log = log;
+    try {
+      wptHookClient = new WptHookClient(url, log);
+    } catch (MalformedURLException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public boolean waitIfNeeded(Command command) throws Exception {
+    boolean hasWaited = false;
+
+    if (command.getName().equals(DriverCommand.QUIT) || command.getName().equals(DriverCommand.CLOSE)) {
+      // we need to inform WPT that the session will terminate
+      wptHookClient.webdriverDone();
```

Comment:
We should probably do a log.info here too just so that we know we dispatched a webdriver_done as part of QUIT.